src/main/java/org/prebid/server/bidder/adprime/AdprimeBidder.java

```diff
@@ -1,20 +1,192 @@
+                .imp(List.of(imp))
+                .build();
+    }
+
+    private User resolveUser(BidRequest bidRequest, ExtImpAdprime extImpAdprime) {
+        final User user = bidRequest.getUser();
+        final List<String> audiences = extImpAdprime.getAudiences();
+
+        if (Objects.isNull(bidRequest.getSite()) || CollectionUtils.isEmpty(audiences)) {
+            return user;
+        }
+
+        final String customData = String.join(",", audiences);
+
+        return Objects.isNull(user)
+                ? User.builder().customdata(customData).build()
+                : user.toBuilder().customdata(customData).build();
+    }
+
+    private Site resolveSite(BidRequest bidRequest, ExtImpAdprime extImpAdprime) {
+        final Site site = bidRequest.getSite();
+        final List<String> keywords = extImpAdprime.getKeywords();
+
+        return Objects.nonNull(site) && CollectionUtils.isNotEmpty(keywords)
+                ? site.toBuilder()
+                .keywords(String.join(",", keywords))
+                .build()
+                : site;
+    }
+
+    private HttpRequest<BidRequest> makeHttpRequest(BidRequest bidRequest) {
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(HttpUtil.headers())
+                .payload(bidRequest)
+                .body(mapper.encode(bidRequest))
+                .build();
+    }
+
+    @Override
+    public final Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final List<BidderError> errors = new ArrayList<>();
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse, errors), errors);
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private static List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse,
+                                               List<BidderError> errors) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidRequest, bidResponse, errors);
+    }
+
+    private static List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse,
+                                                    List<BidderError> errors) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> resolveBidderBid(bid, bidResponse.getCur(), bidRequest.getImp(), errors))
+                .collect(Collectors.toList());
```

Comment:
You have to add `.filter(Objects::nonNull)` since `resolveBidderBid` may return null.